src/map/utils/battleutils.cpp

```diff
@@ -5090,9 +5090,7 @@ namespace battleutils
         }
 
         float chrRatio = ((PCharmer->CHR() - PTarget->CHR())) / 100.f;
```

Comment:
Should make a check here to ensure PCharmer->CHR - PTarget->CHR != 0. Being that attributes are int this could happen a lot, no?

Reply:
There are also double brackets that could be reduced to 1 pair

Reply:
Why would it being 0 / 100 be a problem?

Reply:
So we have diagnosed this issue previously to cause major issues with a large negative value with chrRatio causing charmChance to be fairly consistently low. This clamp was implemented due to the fact that low level BSTs in many cases were not able to even charm low level mobs. The correct fix for the crash would be:

chrRatio = std::clamp(chrRatio, (1 / 100.f), 0.95); // Clamp so we can't have -charmChance

The crash is likely due to the improper clamping of the values as in many cases with low level BSTs we would be looking at a clamp that looks like this:

chrRatio = std::clamp(chrRatio, 0.01, -0.3);

Which this appears to be causing an issue, although error handling "should" in theory handle this.

Reply:
my bad. I had the rule mixed up in my head lol.